python/tk_multi_loader/search_widget.py

```diff
@@ -0,0 +1,128 @@
+
+        # make invisible by default
+        self.setVisible(False)
+        
+        # set up the UI
+        self._ui = Ui_SearchWidget() 
+        self._ui.setupUi(self)
+        
+        # now grab the default background color and use that
+        # in addition to that, apply the same styling that the search
+        # bar in the tree view is using. 
+        p = QtGui.QPalette()
+        bg_col = p.color(QtGui.QPalette.Active, QtGui.QPalette.Window)
+        
+        style = """
+        QGroupBox
+        {
+            background-color: rgb(%s, %s, %s);
+            border-style: none;
+            border-top-left-radius: 0px;
+            border-top-right-radius: 0px;
+            border-bottom-left-radius: 10px;
+            border-bottom-right-radius: 10px;
+        }
+        
+        QLineEdit
+        { 
+            border-width: 1px; 
+            background-image: url(:/res/search.png);
+            background-repeat: no-repeat;
+            background-position: center left;
+            border-radius: 5px; 
+            padding-left:20px;
+            margin:4px;
+            height:22px;
+        }        
+        """ % (bg_col.red(), bg_col.green(), bg_col.blue())
+        
+        self._ui.group.setStyleSheet(style)
+                
+        # hook up a listener to the parent window so this widget
+        # follows along when the parent window changes size
+        filter = ResizeEventFilter(parent)
+        filter.resized.connect(self._on_parent_resized)
+        parent.installEventFilter(filter)
+
+        # set up signals and slots
+        self._ui.search.textChanged.connect(self._on_filter_changed)
+
+    def _on_filter_changed(self):
+        """
+        Callback for when the text changes
+        
+        :param new_text: The contents of the filter line edit box
+        """
+        # emit our custom signal
+        if self.isVisible():
+            # emit the search text that is in the view
+            search_text = self._ui.search.text()
+        else:
+            # widget is hidden - emit empty search text
+            search_text = ""
+        
+        self.filter_changed.emit(search_text)
+
+    def disable(self):
+        """
+        Disable search widget and clear search query.
+        """
+        # hide and reset the search
+        self.setVisible(False)
+        self._on_filter_changed()
+
+    def enable(self):
+        """
+        Enable search widget and focus the keyboard input on it.
+        """
+        self.setVisible(True)
+        self._ui.search.setFocus()
+        self._on_filter_changed()
+    
+    def _on_parent_resized(self):
+        """
+        Special slot hooked up to the event filter.
+        When associated widget is resized this slot is being called.
+        """
+        # offset the position in such a way that it looks like
+        # it is "hanging down" from the adjacent window.
+        self.move(self.parentWidget().width()-self.width()-30,-10)
```

Comment:
it's just offsets chosen after some tweaking! I'll add some comments to explain and I'll make the numbers into proper constants.